### DISCUSSION TITLE
Allow changing observables without actions inside the store definition

### DISCUSSION THREAD

**Discussion by lucas-santosP** [mobxjs/mobx#3105] [Ideas]
An option to `enforceActions` called something like `observedOutsideStoreDefinition`, which allows changing observables inside the store definition without actions, but it would trigger the warning if you try to change an observable outside of the store definition without using an action. 

Is just like the `observed` option, but allowing changes without action inside the store definition. My purpose in asking this is that it would be great to use `async/await` directly inside the store class definition.

Example:
```ts
configure({ enforceActions: "observedOutsideStoreDefinition" });
```

```ts
export class UserStore {
  user: IUser | null;

  constructor() {
    this.user = null;
    makeAutoObservable(this);
  }

  async login(payload: IUser) {
    await waitAsync(); // some async calls
    this.user = payload; // This would be allowed without showing the warning
  }
}
```

**Comment by mweststrate** [comment]
Sadly there is technically no way to detect such scenarios, which is the
reason that flow exists.

**Reply by lucas-santosP** [reply]
Got it. I imagined it wouldn't be possible to check that, but I wanted to ask to be sure.
I'm not a big fan of Generators syntax but I know it has some advantages.

Thanks for the answer.
